### ISSUE TITLE
Completion popup menu not appearing

### ISSUE THREAD

**Issue by pjeschba** [roxma/nvim-completion-manager#55]
Hi roxma. I love the plugin, but after an update the popup menu that populates with a list of autocomplete suggestions is not longer showing up for me. I've never reported an issue on github before - so I'll almost certainly be forgetting some details. If you let me know if any other info would be helpful I'd be happy to provide it! 

[Here's a gif of what's happening:](http://imgur.com/a/B87cg)

basically, you can see that the autocompletion of the def snippet works when I hit tab, but it doesn't appear in the popup menu beforehand. I'm also using Supertab, Ultisnips, and your vim8 compatability extension if that helps at all.

**Comment by roxma** [comment]
Could you follow the trouble shooting section in READNE.md and provide the log of cm_core?

**Comment by pjeschba** [comment]
I ran in my shell:

`NVIM_PYTHON_LOG_FILE=nvim.log NVIM_NCM_LOG_LEVEL=DEBUG NVIM_NCM_MULTI_THREAD=0 nvim bla.py
`
But nothing was generated in my working directory. Did I forget a step?

**Comment by roxma** [comment]
If no log file is generated, you need to start NCM manually and see what's happening, You could still follow the steps in the trouble-shooting section.

**Comment by pjeschba** [comment]
Apologies for the probably simple questions - I'm relatively new to using vim. I was not able to start NCM manually. I ran 

`:echo neovim_rpc#serveraddr()
`
From within vim and the terminal and got [this error](http://imgur.com/a/ecfAW)

**Comment by roxma** [comment]
`NVIM_PYTHON_LOG_FILE=nvim.log NVIM_NCM_LOG_LEVEL=DEBUG NVIM_NCM_MULTI_THREAD=0 nvim bla.py` It seems you're using neovim instead of vim8, in this case you need `:echo v:servername` instead of `:echo neovim_rpc#serveraddr()`

If you are using vim8, than the error you provided should be the cause of this issue.

**Comment by pjeschba** [comment]
I am using vim8, not neovim. Sorry for the above, it was a typo in what I posted on GitHub. I ran that command with vim, not neovim.

**Comment by roxma** [comment]
Looks like the vim-hug-neovim-rpc fails to execute `:python3 import neovim_rpc_server` or `:python import neovim_rpc_server`. You could check that manually.

Also make sure the [pythonx](https://github.com/roxma/vim-hug-neovim-rpc/tree/master/pythonx) directory of vim-hug-neovim-rpc is in the installation directory.

**Comment by pjeschba** [comment]
It says there is "No module named neovim_rpc_server" when I ran the latter command. I double checked to ensure my vim version has python as well.

The pythonx directory is in .vim/bundle/vim-hug-neovim-rpc/; I installed the plugins with Vundle.

**Comment by roxma** [comment]
If vim failed to execute both  both `:python3 import neovim_rpc_server` and `:python import neovim_rpc_server`.

Use `:echo globpath(&rtp,'pythonx/neovim_rpc_server.py',1)` to check the installation of vim-hug-neovim-rpc. If the result is empty, then vim-hug-neovim-rpc is somehow not  properly installed. Something wrong with rtp maybe.

**Comment by pjeschba** [comment]
That returned this:

"/Users/<username>/.vim/bundle/vim-hug-neovim-rpc/pythonx/neovim_rpc_server.py"

**Comment by roxma** [comment]
This is really wierd. Vim did successfully load the autoload function, but the python fails to search the pythonx directory.

`:help pythonx` may give you some useful  information.

**Comment by roxma** [comment]
Maybe there's some other error going on.

Remove NCM, leave vim-hug-neovim-rpc installed alone.

Then execute `:python import neovim_rpc_server` and check the error message.

**Comment by pjeschba** [comment]
I get the same "No module named neovim_rpc_server" error with just vim-hug-neovim-rpc installed.

**Comment by roxma** [comment]
Also make sure vim can successfully execute `:python import neovim`, this is a new dependency added  in recent updates.

**Comment by roxma** [comment]
I think this is it, vim also reports "No module named neovim_rpc_server"  when dependency is not satisfied.

**Comment by roxma** [comment]
I'll add some verbose information when this error uccor, thanks for your information.

**Comment by pjeschba** [comment]
Ahhh, that would make sense. I can’t execute that command and did not even have neovim installed. I just installed neovim via home-brew, but am still getting an error when running “:python import neovim” currently. Do I have to point something at my new neovim installation?

**Comment by roxma** [comment]
Vim seems not to choose the python installed via homebrew, 

this might be useful: http://vi.stackexchange.com/questions/6604/how-to-access-a-custom-python-module-inside-vim

**Comment by roxma** [comment]
I guess that execute `pip install neovim` for your system wide python instead of the one installed via home-brew, would fix this error.

**Comment by pjeschba** [comment]
It works now! Thank you so much for your help.
